airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/TypeAndDedupeQueryBuilder.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.base.destination.typing_deduping;
+
+import io.airbyte.protocol.models.v0.DestinationSyncMode;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+public interface TypeAndDedupeQueryBuilder {
```

Comment:
compare with SqlGenerator

Reply:
For time being I removed this dependency. Once i get more granular queries (if needed) will add it as a composition